### ISSUE TITLE
When starting Exams, Notifications and Attempt logs can be out of sync

### ISSUE THREAD

**Issue by jonboiser** [learningequality/kolibri#4898]
### Observed behavior

1. As a learner, start an exam, but do not do any of the questions
1. After some time, a Notification will be recorded for the exam with a status of "Started"
1. However, querying the class summary API, the 'exam learner status' for that exam will have a status of "NotStarted".

This can lead to an inconsistency, where the Notification will update the in-memory classSummary data and cause the dashboard and reports to show that 1 learner has "started" an exam, but if you were to refresh (and get the on-server class summary data without the updating notification), it will revert to showing 0 learners starting the exam.

### Expected behavior

Since, the UI intends to use Notifications to patch class summary in real time, the two notions of "Exam Started" should match to avoid situations like the on described above.

### User-facing consequences

inconsistent / fluctuating values in reports

### Steps to reproduce

…

### Context

0.12.0 a 7

**Comment by indirectlylit** [comment]
cc @rtibbles @jredrejo 

In `class_summary_api` I compute the status as

```python
    def get_status(self, exam_log):
        if exam_log.closed:
            return COMPLETED
        elif exam_log.attemptlogs.values_list("item").count() > 0:
            return STARTED
        return NOT_STARTED
```

Any recommendations on how to update this to be consistent with the notification behavior?

**Comment by rtibbles** [comment]
I guess similar to the `needs_help` work I did in the same API, querying the notifications to do the `STARTED` flagging?

**Comment by indirectlylit** [comment]
Is there not information in the exam logs about whether the user has opened it or not? That's all I need here

**Comment by jamalex** [comment]
The first time the user opens the exam, it creates the ExamLog itself. I'll change the logic so it doesn't check for attemptlogs, just returns started if there's an ExamLog at all.

**Comment by indirectlylit** [comment]
perfect

**Comment by jamalex** [comment]
Do we have the same issue for other types of content? In `content_status_serializer`, I see we do:
```
        elif log["progress"] == 0:
            return NOT_STARTED
```

Do we want it to show as "started" as soon as people open other types of content, or only once `progress > 0`?

**Comment by jamalex** [comment]
For now, just fixed it for exams in #4903.

**Comment by indirectlylit** [comment]
> Do we want it to show as "started" as soon as people open other types of content, or only once `progress > 0`?

The current behavior is still correct for content, because it allows people to navigate around without looking like they've done a bunch of work

quizzes seem a little different because it's like "turning over the paper" and taking a look at the questions
